src/main/scala/org/hammerlab/guacamole/pileup/Pileup.scala

```diff
@@ -150,8 +150,7 @@ object Pileup {
    * @return A [[Pileup]] at the given locus.
    */
   def apply(reads: Seq[MappedRead], referenceName: String, locus: Long, referenceContigSequence: ContigSequence): Pileup = {
-    //TODO: Is this call to overlaps locus necessary?
-    val elements = reads.filter(_.overlapsLocus(locus)).map(PileupElement(_, locus, referenceContigSequence))
+    val elements = reads.map(PileupElement(_, locus, referenceContigSequence))
```

Comment:
I think if you remove the filter then this doc string:

```
@param reads Sequence of reads, in any order, that may or may not overlap the locus.
```

Should change to something like "Sequence of reads, in any order, that must overlap the locus."

Were you seeing overhead in all the `overlaps` checks?